server/resourcemanager/testutil/operations_list.go

```diff
@@ -155,29 +163,28 @@ func buildPaginationOperationStep(sortDirection, sortField string) operationTest
 			jsonBody := responseBodyJSON(t, resp, ct)
 
 			var parsedJsonBody struct {
-				Count int              `json:count`
-				Items []map[string]any `json:items`
+				Count int              `json:"count"`
+				Items []map[string]any `json:"items"`
 			}
 			json.Unmarshal([]byte(jsonBody), &parsedJsonBody)
 
 			require.Equal(t, 3, parsedJsonBody.Count)
+			require.Greater(t, len(parsedJsonBody.Items), 1)
 
 			var prevVal any
-			field := rt.sortFields[0]
-
 			for _, item := range parsedJsonBody.Items {
+				itemSpec := item["spec"].(map[string]any)
```

Comment:
this test was never doing anything because `item[field]` was always nil, because the structure was incorrectly parsed.

Reply:
Wondering if it makes sense: 
We are assuming the structure of the `Resource[T ResourceSpec]` here, but usually, we want to work only with the `ResourceSpec`. Makes sense to have "wrap/unwrap" function in our tests to make this behavior explicit? 

Why? We can use it here to "unwrap" resource and test it and also use it on `SampleJSON` and `SampleJSONUpdated` to pass the "raw" resource as a go object, and later convert it to JSON and wrap it on a `Resource` json.

Does it make sense?

Reply:
THe idea of these tests is to "black box" test as much as possible of a resource. The `Resource[T]` is not something we assume, but a standard format for the resourcemanager. all api calls/responses wrap around that. The tests must be as "from the user perspective" as possible, so we don't hide any behaviours